HopsanGenerator/templates/simulinkWrapperTemplate.cpp

```diff
@@ -94,25 +84,25 @@ static void mdlInitializeSizes(SimStruct *S)
         ssSetErrorStatus(S,"Error could not open model: <<<4>>>");
         return;
     }
-    startT = ssGetTStart(S);
-    stopT = ssGetTFinal(S);
     pComponentSystem->setDesiredTimestep(<<<timestep>>>);
+
 <<<5>>>}
 
 
 static void mdlInitializeSampleTimes(SimStruct *S)
 {
-    //ssSetSampleTime(S, 0, 0.001);
-    //ssSetOffsetTime(S, 0, 0.0);
+    ssSetSampleTime(S, 0, 0.001);
```

Comment:
I am unsure how this works, is 0.001 the default step time ? 
I assume that the user can change this from simulation somehow ?

Reply:
It is actually hard-coded, depending on the step size in the exported Hopsan model. Making this a tunable parameter is not possible, because sample time is defined only once before reading any other variables in a Simulink simulation. We used to have this commented out, but I noticed it could cause problems if running Simulink with variable step-size. As I interpret it, this will force Simulink to call this block at exactly the specified interval. Not perfect, but it should not be worse than before either.